code/modules/clothing/chameleon.dm

```diff
@@ -760,3 +763,167 @@
 /obj/item/clothing/neck/chameleon/broken/Initialize(mapload)
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+
+/obj/item/gun/energy/laser/chameleon
+	var/datum/action/item_action/chameleon/change/chameleon_action
+
+	ammo_type = list(/obj/item/ammo_casing/energy/chameleon)
+	pin = /obj/item/firing_pin
+	automatic_charge_overlays = FALSE
+	can_select = FALSE
+
+	/// The vars copied over to our projectile on fire.
+	var/list/chameleon_projectile_vars
+
+	/// The badmin mode. Makes your projectiles act like the real deal.
+	var/real_hits
+
+/obj/item/gun/energy/laser/chameleon/New()
```

Comment:
Why is this `New` and not `Initialize`?

If this is intentional, this needs a comment, since it currently looks (and probably is) incorrect.